### PR TITLE
Simplified a lot of code through Resharper

### PR DESCRIPTION
**Pull request by robinlavallee**

### CODE REVIEW

File: unity/Assets/Scripts/Game/Managers/EntityManager.cs [hunk 1]
```diff
@@ -30,8 +29,9 @@ private void UnregisterPlayer(Player player_)
 
     public int Register(Entity entity_)
     {
-        if (entity_ is Player)
```
Comment (jstine35) [inline unity/Assets/Scripts/Game/Managers/EntityManager.cs]: Hmm, not sure Resharper was helpful in this specific case. I prefer the original when it's a simple one-off use of an object like this.

Comment (robinlavallee) [reply unity/Assets/Scripts/Game/Managers/EntityManager.cs]: Yea, I see. I like the fact that it avoids two casts and just do one. But in the second version, it introduces a new variable.

File: unity/Assets/Scripts/Game/Managers/EntityManager.cs [hunk 2]
```diff
@@ -44,8 +44,9 @@ public void Unregister(Entity entity_)
         //Debug.Assert(m_entities.Contains(entity_), String.Format("Entity {0} is not registered.", entity_));
         m_entities.Remove(entity_);
 
-        if (entity_ is Player)
```
[the remark this answers is not in the document]
Comment (robinlavallee) [reply unity/Assets/Scripts/Game/Managers/EntityManager.cs]: It is one of the rule, collapsing two casts.

**Comment by robinlavallee** [comment]
Replaced with #142

**Pull request closed without merging**